src-java/kilda-pce/src/main/java/org/openkilda/pce/impl/InMemoryPathComputer.java

```diff
@@ -153,9 +156,27 @@ private FindPathResult findPathInNetwork(Flow flow, AvailableNetwork network,
                 return pathFinder.findPathWithMinWeightAndLatencyLimits(network, flow.getSrcSwitchId(),
                         flow.getDestSwitchId(), weightFunction, maxLatency, maxLatencyTier2);
             case MAX_LATENCY:
-                return pathFinder.findPathWithWeightCloseToMaxWeight(network, flow.getSrcSwitchId(),
-                        flow.getDestSwitchId(), weightFunction, flow.getMaxLatency(),
-                        Optional.ofNullable(flow.getMaxLatencyTier2()).orElse(0L));
+                try {
+                    return pathFinder.findPathWithWeightCloseToMaxWeight(network, flow.getSrcSwitchId(),
+                            flow.getDestSwitchId(), weightFunction, flow.getMaxLatency(),
+                            Optional.ofNullable(flow.getMaxLatencyTier2()).orElse(0L));
+                } catch (UnroutableFlowException e) {
+                    if (e.getFailReason() == null
+                            || !e.getFailReason().containsKey(FailReasonType.MAX_WEIGHT_EXCEEDED)
+                            || e.getFailReason().get(FailReasonType.MAX_WEIGHT_EXCEEDED).getWeight() == null
+                            || !e.getMessage().contains("Reasons")) {
+                        throw e;
+                    }
+                    String[] split = e.getMessage().split("Reasons");
+                    long actualLatency = e.getFailReason().get(FailReasonType.MAX_WEIGHT_EXCEEDED).getWeight();
+                    Map<FailReasonType, FailReason> reasons = e.getFailReason();
+                    reasons.put(FailReasonType.MAX_WEIGHT_EXCEEDED, new FailReason(FailReasonType.MAX_WEIGHT_EXCEEDED,
```

Comment:
Is it an expected result to have `Reasons: Max weight exceeded: Requested path must have latency 10000000 or lower, but best path has latency 66666666`? What max weight has common with max latency?
Also, please, specify units in error message if possible. 
In this particular case user creates flow with `.setMaxLatency(10L)` (I suspect it's milliseconds, but can't find a description anywhere), but in error response I see 10 millions of nanoseconds(?).

Reply:
This is because, inside the findPathWithWeightCloseToMaxWight, we don't know if the weight is latency or not, so there, the latency=weight. But I can change the last message to save the reason as Latency directly.

In getNPaths there is a conversion from Duration to Long in nanoseconds. So I guess this is the unit the PCE is managing. There must be another place out of PCE that makes a similar conversion when it creates the Flow object.

So, I think the best option is just to say that it is nanosecond instead of doing another conversion. What do you think?

Reply:
So, correct me if I'm wrong. 
If we try to create a flow with `max_latency = <something>` and path computation strategy `MAX_LATENCY`, latency becomes weight and we receive error `MAX_WEIGHT_EXCEEDED`.
If we try to create flow with any other computation strategy, e.g. `COST` or `LATENCY`, we don't consider latency as the weight or we ignore `max_latency` parameter. So we can't receive `LATENCY_LIMIT` exception.
Thus, we don't need `LATENCY_LIMIT` exception reason, because it would never be used?

Reply:
Yes because in this other case we actually know that is latency.

https://github.com/telstra/open-kilda/blob/a2af54a4d841bd3ec5656e3946e86789ad376993/src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java#L348